test/new-e2e/utils/e2e/client/docker.go

```diff
@@ -57,20 +61,30 @@ func (docker *Docker) initService(t *testing.T, data *docker.ClientData) error {
 	}
 
 	docker.client, err = client.NewClientWithOpts(opts...)
+	if docker.agentContainerName != "" {
+		docker.agent = newAgentCommandRunner(t, &dockerAgentCommand{docker: docker})
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
It feels weird to define an `agent` instance that has a reference to its parent, could `newAgentCommandRunner` accept an `executeCommand` function instead, of type 

```golang
type executeAgentCommand func(commands []string) (string, error)
```
So that instead of defining extra structs you could define a function here passing the `docker` reference ?

I think it would make code more readable, currently there are many levels of `commandRunner`

Reply:
>It feels weird to define an agent instance that has a reference to its parent
I don't think it is weird, think about a graph.

> could newAgentCommandRunner accept an executeCommand function instead
I like this idea as it remove struct like `dockerAgentRawCommandRunner` (The naming is not so great).